design/notifications.md

```diff
@@ -0,0 +1,360 @@
+      description: The name of the branch to transplant commits into.
+    expectedHash:
+      type: string
+      description: The hash at which the branch was expected to be in at the moment of the transplant.
+    newHash:
+      type: string
+      description: The hash of the head of this branch after the transplant.
+
+```
+
+#### Example
+```json
+{
+  "type": "TRANSPLANT",
+  "eventTime": "2009-02-14T18:09:10Z",
+  "fromRefName": "dev",
+  "fromHashes": ["d440310c007bb303", "46d1ec0e7680b0ed", "018eb26ce9eb4712"],
+  "toBranchName": "main",
+  "expectedHash": "85a1090b5fe01a7b",
+  "newHash": "01f53acc6fdd6134"
+}
+```
+
+### Reference creation
+Reference creation events are triggered whenever a reference is created. In other words, whenever the 
+`POST /trees/tree/` API in Nessie is called.
+
+#### Schema
+```yaml
+ReferenceCreated:
+  type: object
+  properties:
+    type:
+      type: string
+      description: Always REFERENCE_CREATED
+    eventTime:
+      type: string
+      format: date
+      description: The time the event is fired in ISO date format.
+    reference:
+      $ref: Reference
+      description: The recently created reference.
+```
+
+#### Example
+```json
+{
+  "type": "REFERENCE_CREATED",
+  "eventTime": "2009-02-14T18:09:10Z",
+  "reference": {
+    "type":"BRANCH",
+    "name":"dev",
+    "hash":"e249ff3a681f177c"
+  }
+}
+```
+
+### Reference assignment
+Reference assignment events are triggered whenever a reference is assigned. In other words, whenever the 
+`PUT /trees/{referenceType}/{referenceName}` API in Nessie is called.
+
+#### Schema
+```yaml
+ReferenceAssigned:
+  type: object
+  properties:
+    type:
+      type: string
+      description: Always REFERENCE_ASSIGNED
+    eventTime:
+      type: string
+        format: date
+        description: The time the event is fired in ISO date format.
+    reference:
+      $ref: Reference
+      description: The reassigned reference.
+    assignedTo:
+      $ref: Reference 
+      description: The reference to which the reference got assigned to.
+```
+
+#### Example
+```json
+{
+  "type": "REFERENCE_ASSIGNED",
+  "eventTime": "2009-02-14T18:09:10Z",
+  "reference": {
+    "type": "BRANCH",
+    "name": "dev",
+    "hash": "e249ff3a681f177c"
+  },
+  "assignedTo": {
+    "type": "TAG",
+    "name": "release-10",
+    "hash": "e249ff3a681f177c"
+  }
+}
+```
+
+### Reference deletion
+Reference deletion events are triggered whenever a reference is deleted. In other words,  whenever the 
+`DELETE /trees/{referenceType}/{referenceName}` API in Nessie is called.
+
+#### Schema
+```yaml
+ReferenceDeleted:
+  type: object
+  properties:
+    type:
+      type: string
+      description: Always REFERENCE_DELETED
+    eventTime:
+      type: string
+      format: date
+      description: The time the event is fired in ISO date format.
+    referenceType:
+      type: string
+      description: The reference type. Either BRANCH or TAG.
+    referenceName:
```

Comment:
we include the hash in a lot of other places, should we include it here? eg the old hash for an assign feels like the same category

Reply:
Will change this to be the reference being deleted.